### PR TITLE
fix(seed): Best practice for radio button

### LINKED ISSUE #15946: Create a Set of Radio Buttons - Not Showing Best Practices
In the *Create a Set of Radio Buttons* challenge of the *HTML5 and CSS* section (https://www.freecodecamp.org/challenges/create-a-set-of-radio-buttons), it states:

> Here's an example of a radio button:
> ```
> <label><input type="radio" name="indoor-outdoor"> Indoor</label>
> ```

Yet, in more official Mozilla documentation (https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/How_to_structure_an_HTML_form#The_%3Clabel%3E_element), it states:

> Note that a widget can be nested inside its <label> element, like so:
> 
> ```
> <label for="name">
>   Name: <input type="text" id="name" name="user_name">
> </label>
> ```
> 
> Even in such cases however, it is considered best practice to set the for attribute because some assistive technologies do not understand implicit relationships between labels and widgets.

This was discussed here: https://forum.freecodecamp.org/t/anyone-notice-that-some-of-the-challenges-dont-demonstrate-best-practices/151348

Please update the challenge to reflect best practices.

### PR DESCRIPTION
**Pull request by darthm3gatron**
Closes #15946

#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [x] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #XXXXX

### CODE REVIEW

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 1]
```diff
@@ -1769,9 +1769,10 @@
...
         "All related radio buttons should have the same <code>name</code> attribute.",
         "Here's an example of a radio button:",
-        "<code>&#60;label&#62;&#60;input type=\"radio\" name=\"indoor-outdoor\"&#62; Indoor&#60;/label&#62;</code>",
+        "<code>&#60;label&#62;&#60;input type=\"radio\" name=\"indoor-outdoor\" for=\"indoor\" &#62; Indoor&#60;/label&#62;</code>",
         "<hr>",
-        "Add a pair of radio buttons to your form. One should have the option of <code>indoor</code> and the other should have the option of <code>outdoor</code>. Both should share the <code>name</code> attribute of <code>indoor-outdoor</code>."
+        "Add a pair of radio buttons to your form. One should have the option of <code>indoor</code> and the other should have the option of <code>outdoor</code>. Both should share the <code>name</code> attribute of <code>indoor-outdoor</code>.",
+        "It is considered best practice to set the for attribute for assistive technologies to be able to link the relationship between the label and the child elements such as an input in this case."
```
Comment (raisedadead) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Please wrap the `for` and `input` around `<code>` tags.

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
Also please fill the checklist in the PR description above

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 2]
```diff
@@ -1769,9 +1769,10 @@
         "Each of your radio buttons should be nested within its own <code>label</code> element.",
         "All related radio buttons should have the same <code>name</code> attribute.",
         "Here's an example of a radio button:",
-        "<code>&#60;label&#62;&#60;input type=\"radio\" name=\"indoor-outdoor\"&#62; Indoor&#60;/label&#62;</code>",
+        "<code>&#60;label&#62;&#60;input type=\"radio\" name=\"indoor-outdoor\" for=\"indoor\" &#62; Indoor&#60;/label&#62;</code>",
```
Comment (raisedadead) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Some more changes:

1. The code sample needs the `for` attribute inside the `<label>`, NOT the `<input>`
2. Lets, split it into a block to make it cleaner. (refer: [Challenge Style Guide](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/seed/challenge-style-guide.md))
3. The clarification text is to be placed after the code sample.

```diff
-        "<code>&#60;label&#62;&#60;input type=\"radio\" name=\"indoor-outdoor\" for=\"indoor\" &#62; Indoor&#60;/label&#62;</code>",
+        "<blockquote>&#60;label for=\"indoor\"&#62; <br> &#60;input type=\"radio\" name=\"indoor-outdoor\"&#62;Indoor <br>&#60;/label&#62;</blockquote>",
+        "It is considered best practice to set the <code>for</code> attribute for assistive technologies to be able to link the relationship between the label and the child elements such as an <code>input</code> in this case.",
```

File: seed/challenges/01-responsive-web-design/basic-html-and-html5.json [hunk 3]
```diff
@@ -1769,9 +1769,10 @@
...
         "<hr>",
-        "Add a pair of radio buttons to your form. One should have the option of <code>indoor</code> and the other should have the option of <code>outdoor</code>. Both should share the <code>name</code> attribute of <code>indoor-outdoor</code>."
+        "Add a pair of radio buttons to your form. One should have the option of <code>indoor</code> and the other should have the option of <code>outdoor</code>. Both should share the <code>name</code> attribute of <code>indoor-outdoor</code>.",
+        "It is considered best practice to set the <code>for</code> attribute for assistive technologies to be able to link the relationship between the label and the child elements such as an <code>input</code> in this case."
```
Comment (raisedadead) [inline seed/challenges/01-responsive-web-design/basic-html-and-html5.json]: Remove this line

```diff
-        "It is considered best practice to set the <code>for</code> attribute for assistive technologies to be able to link the relationship between the label and the child elements such as an <code>input</code> in this case."
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
@darthm3gatron nice work, some finishing touches and we are good to go!

**Comment by darthm3gatron** [comment]
It's telling me the Travis CI build failed. I'm not sure what went wrong there.

**Comment by raisedadead** [comment]
There is an invalid `+` character in the new changes. This is also easily caught when you run tests and locally check by running the local server.

```
npm run test
```

```
npm run develop
```

**Comment by darthm3gatron** [comment]
Sorry about that. I'm new to all this.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **LGTM.** :sparkles: :sparkles: 

Real nice work! Sorry about so many change requests, its just a part of reviewing. 

Thanks @darthm3gatron,  for this pull request and contributing to the code-base for the first time. 

Happy contributing!

**Pull request merged**